client/post-editor/media-modal/index.jsx

```diff
@@ -210,7 +210,7 @@ module.exports = React.createClass( {
 		this.setView( ModalViews.IMAGE_EDITOR );
 	},
 
-	onImageEditorDone( error, blob, imageEditorProps ) {
+	onImageEditorDone( error, blob, imageEditorProps, media ) {
```

Comment:
Media id can be an optional prop to pass back. If we have it we send it to the callback, if not,  we don't. The alternative would be to get the currently selected media from this onImageEditorDone instance. (Which we should have hooks to in the media modal)